README.md
[comment]: <> (# [Material Dashboard Angular - Free Bootstrap Material Design Admin]&#40;https://www.creative-tim.com/product/material-dashboard-angular2&#41;[![version][version-badge]][CHANGELOG])

[comment]: <> (![Product Gif]&#40;https://raw.githubusercontent.com/creativetimofficial/public-assets/master/material-dashboard-angular/material-dashboard-angular.gif&#41;)

[comment]: <> (**[Material Dashboard Angular]&#40;https://www.creative-tim.com/product/material-dashboard-angular2/&#41;** is a free Material Bootstrap Admin with a fresh, new design inspired by Google's Material Design. We are very excited to introduce our take on the material concepts through an easy to use and beautiful set of components. Material Dashboard was built over the popular Bootstrap framework and it comes with a couple of third-party plugins redesigned to fit in with the rest of the elements.)

[comment]: <> (Material Dashboard makes use of light, surface and movement. The general layout resembles sheets of paper following multiple different layers, so that the depth and order is obvious. The navigation stays mainly on the left sidebar and the content is on the right inside the main panel.)

[comment]: <> (This product came as a result of users asking for a material dashboard after we released our successful [Material Kit]&#40;http://www.creative-tim.com/product/material-kit&#41;. We developed it based on your feedback and it is a powerful bootstrap admin dashboard, which allows you to build products like admin panels, content managements systems and CRMs.)

[comment]: <> (Material Dashboard comes with 5 color filter choices for both the sidebar and the card headers &#40;blue, green, orange, red and purple&#41; and an option to have a background image on the sidebar.)

[comment]: <> (Material Dashboard uses a framework built by our friend [Federico - Bootstrap Material Design]&#40;http://fezvrasta.github.io/bootstrap-material-design/&#41;, who did an amazing job creating the backbone for the material effects, animations, ripples and transitions. Big thanks to his team for the effort and forward thinking they put into it.)

[comment]: <> (Special thanks go to:)

[comment]: <> ([Robert McIntosh]&#40;https://github.com/mouse0270/bootstrap-notify&#41; for the notification system.)

[comment]: <> ([Chartist]&#40;https://gionkunz.github.io/chartist-js/&#41; for the wonderful charts.)

[comment]: <> (We are very excited to share this dashboard with you and we look forward to hearing your feedback!)

[comment]: <> (You can find the Github Repo here.)

[comment]: <> (## Table of Contents)

[comment]: <> (* [Versions]&#40;#versions&#41;)

[comment]: <> (* [Demo]&#40;#demo&#41;)

[comment]: <> (* [Quick Start]&#40;#quick-start&#41;)

[comment]: <> (* [Documentation]&#40;#documentation&#41;)

[comment]: <> (* [File Structure]&#40;#file-structure&#41;)

[comment]: <> (* [Browser Support]&#40;#browser-support&#41;)

[comment]: <> (* [Resources]&#40;#resources&#41;)

[comment]: <> (* [Reporting Issues]&#40;#reporting-issues&#41;)

[comment]: <> (* [Technical Support or Questions]&#40;#technical-support-or-questions&#41;)

[comment]: <> (* [Licensing]&#40;#licensing&#41;)

[comment]: <> (* [Useful Links]&#40;#useful-links&#41;)


[comment]: <> (## Versions)

[comment]: <> ([<img src="https://github.com/creativetimofficial/public-assets/blob/master/logos/html-logo.jpg?raw=true" width="60" height="60" />]&#40;https://www.creative-tim.com/product/material-dashboard&#41;)

[comment]: <> ([<img src="https://github.com/creativetimofficial/public-assets/blob/master/logos/angular-logo.jpg?raw=true" width="60" height="60" />]&#40;https://www.creative-tim.com/product/material-dashboard-angular2&#41;)

[comment]: <> ([<img src="https://github.com/creativetimofficial/public-assets/blob/master/logos/vue-logo.jpg?raw=true" width="60" height="60" />]&#40;https://www.creative-tim.com/product/vue-material-dashboard&#41;)

[comment]: <> ([<img src="https://github.com/creativetimofficial/public-assets/blob/master/logos/react-logo.jpg?raw=true" width="60" height="60" />]&#40;https://www.creative-tim.com/product/material-dashboard-react&#41;)

[comment]: <> (| HTML | Angular | Vue | React |)

[comment]: <> (| --- | --- | --- | --- |)

[comment]: <> (| [![Material Dashboard HTML]&#40;https://github.com/creativetimofficial/public-assets/blob/master/material-dashboard-html/material-dashboard.jpg?raw=true&#41;]&#40;https://www.creative-tim.com/product/material-dashboard&#41; | [![Material Dashboard Angular]&#40;https://github.com/creativetimofficial/public-assets/blob/master/material-dashboard-angular/material-dashboard-angular.jpg?raw=true&#41;]&#40;https://www.creative-tim.com/product/material-dashboard-angular2&#41; | [![Vue Material Dashboard ]&#40;https://github.com/creativetimofficial/public-assets/blob/master/vue-material-dashboard/vue-material-dashboard.jpg?raw=true&#41;]&#40;https://www.creative-tim.com/product/vue-material-dashboard&#41; | [![Material Dashboard React]&#40;https://github.com/creativetimofficial/public-assets/blob/master/material-dashboard-react/material-dashboard-react.jpg?raw=true&#41;]&#40;https://www.creative-tim.com/product/material-dashboard-react&#41;)

[comment]: <> (## Demo)

[comment]: <> (| Dashboard | User Profile | Tables | Icons | Notifications |)

[comment]: <> (| --- | --- | --- | --- | --- |)

[comment]: <> (| [![Start page]&#40;https://raw.githubusercontent.com/creativetimofficial/public-assets/master/material-dashboard-angular/dashboard.png?raw=true&#41;]&#40;https://demos.creative-tim.com/material-dashboard-angular2/#/dashboard&#41; | [![User profile page]&#40;https://raw.githubusercontent.com/creativetimofficial/public-assets/master/material-dashboard-angular/user-profile.png?raw=true&#41;]&#40;https://demos.creative-tim.com/material-dashboard-angular2/#/user-profile&#41; | [![Tables page ]&#40;https://raw.githubusercontent.com/creativetimofficial/public-assets/master/material-dashboard-angular/tables.png?raw=true&#41;]&#40;https://demos.creative-tim.com/material-dashboard-angular2/#/table-list&#41; | [![Icons Page]&#40;https://raw.githubusercontent.com/creativetimofficial/public-assets/master/material-dashboard-angular/icons.png?raw=true&#41;]&#40;https://demos.creative-tim.com/material-dashboard-angular2/#/maps&#41; | [![Notifications page]&#40;https://raw.githubusercontent.com/creativetimofficial/public-assets/master/material-dashboard-angular/notifications.png?raw=true&#41;]&#40;https://demos.creative-tim.com/material-dashboard-angular2/#/notifications&#41;)

[comment]: <> ([View More]&#40;https://demos.creative-tim.com/material-dashboard-angular2/#/dashboard&#41;.)

[comment]: <> (## Quick start)

[comment]: <> (Quick start options:)

[comment]: <> (- [Download from Github]&#40;https://github.com/tiniestory/material-dashboard-angular2/archive/master.zip&#41;.)

[comment]: <> (- [Download from Creative Tim]&#40;http://www.creative-tim.com/product/material-dashboard-angular2&#41;.)

[comment]: <> (## Terminal Commands)

[comment]: <> (This project was generated with [Angular CLI]&#40;https://github.com/angular/angular-cli&#41; version 1.0.0 and angular 4.x.)

[comment]: <> (1. Install NodeJs from [NodeJs Official Page]&#40;https://nodejs.org/en&#41;.)

[comment]: <> (2. Open Terminal)

[comment]: <> (3. Go to your file project)

[comment]: <> (4. Make sure you have installed [Angular CLI]&#40;https://github.com/angular/angular-cli&#41; already. If not, please install.)

[comment]: <> (5. Run in terminal: ```npm install```)

[comment]: <> (6. Run `ng serve` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.)

[comment]: <> (To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI README]&#40;https://github.com/angular/angular-cli/blob/master/README.md&#41;.)

[comment]: <> (### What's included)

[comment]: <> (Within the download you'll find the following directories and files:)

[comment]: <> (```)

[comment]: <> (material-dashboard-angular)

[comment]: <> (├── CHANGELOG.md)

[comment]: <> (├── LICENSE.md)

[comment]: <> (├── README.md)

[comment]: <> (├── angular-cli.json)

[comment]: <> (├── documentation)

[comment]: <> (├── e2e)

[comment]: <> (├── karma.conf.js)

[comment]: <> (├── package-lock.json)

[comment]: <> (├── package.json)

[comment]: <> (├── protractor.conf.js)

[comment]: <> (├── src)

[comment]: <> (│   ├── app)

[comment]: <> (│   │   ├── app.component.css)

[comment]: <> (│   │   ├── app.component.html)

[comment]: <> (│   │   ├── app.component.spec.ts)

[comment]: <> (│   │   ├── app.component.ts)

[comment]: <> (│   │   ├── app.module.ts)

[comment]: <> (│   │   ├── app.routing.ts)

[comment]: <> (│   │   ├── components)

[comment]: <> (│   │   │   ├── components.module.ts)

[comment]: <> (│   │   │   ├── footer)

[comment]: <> (│   │   │   │   ├── footer.component.css)

[comment]: <> (│   │   │   │   ├── footer.component.html)

[comment]: <> (│   │   │   │   ├── footer.component.spec.ts)

[comment]: <> (│   │   │   │   └── footer.component.ts)

[comment]: <> (│   │   │   ├── navbar)

[comment]: <> (│   │   │   │   ├── navbar.component.css)

[comment]: <> (│   │   │   │   ├── navbar.component.html)

[comment]: <> (│   │   │   │   ├── navbar.component.spec.ts)

[comment]: <> (│   │   │   │   └── navbar.component.ts)

[comment]: <> (│   │   │   └── sidebar)

[comment]: <> (│   │   │       ├── sidebar.component.css)

[comment]: <> (│   │   │       ├── sidebar.component.html)

[comment]: <> (│   │   │       ├── sidebar.component.spec.ts)

[comment]: <> (│   │   │       └── sidebar.component.ts)

[comment]: <> (│   │   ├── dashboard)

[comment]: <> (│   │   │   ├── dashboard.component.css)

[comment]: <> (│   │   │   ├── dashboard.component.html)

[comment]: <> (│   │   │   ├── dashboard.component.spec.ts)

[comment]: <> (│   │   │   └── dashboard.component.ts)

[comment]: <> (│   │   ├── icons)

[comment]: <> (│   │   │   ├── icons.component.css)

[comment]: <> (│   │   │   ├── icons.component.html)

[comment]: <> (│   │   │   ├── icons.component.spec.ts)

[comment]: <> (│   │   │   └── icons.component.ts)

[comment]: <> (│   │   ├── layouts)

[comment]: <> (│   │   │   └── admin-layout)

[comment]: <> (│   │   │       ├── admin-layout.component.html)

[comment]: <> (│   │   │       ├── admin-layout.component.scss)

[comment]: <> (│   │   │       ├── admin-layout.component.spec.ts)

[comment]: <> (│   │   │       ├── admin-layout.component.ts)

[comment]: <> (│   │   │       ├── admin-layout.module.ts)

[comment]: <> (│   │   │       └── admin-layout.routing.ts)

[comment]: <> (│   │   ├── maps)

[comment]: <> (│   │   │   ├── maps.component.css)

[comment]: <> (│   │   │   ├── maps.component.html)

[comment]: <> (│   │   │   ├── maps.component.spec.ts)

[comment]: <> (│   │   │   └── maps.component.ts)

[comment]: <> (│   │   ├── notifications)

[comment]: <> (│   │   │   ├── notifications.component.css)

[comment]: <> (│   │   │   ├── notifications.component.html)

[comment]: <> (│   │   │   ├── notifications.component.spec.ts)

[comment]: <> (│   │   │   └── notifications.component.ts)

[comment]: <> (│   │   ├── table-list)

[comment]: <> (│   │   │   ├── table-list.component.css)

[comment]: <> (│   │   │   ├── table-list.component.html)

[comment]: <> (│   │   │   ├── table-list.component.spec.ts)

[comment]: <> (│   │   │   └── table-list.component.ts)

[comment]: <> (│   │   ├── typography)

[comment]: <> (│   │   │   ├── typography.component.css)

[comment]: <> (│   │   │   ├── typography.component.html)

[comment]: <> (│   │   │   ├── typography.component.spec.ts)

[comment]: <> (│   │   │   └── typography.component.ts)

[comment]: <> (│   │   ├── upgrade)

[comment]: <> (│   │   │   ├── upgrade.component.css)

[comment]: <> (│   │   │   ├── upgrade.component.html)

[comment]: <> (│   │   │   ├── upgrade.component.spec.ts)

[comment]: <> (│   │   │   └── upgrade.component.ts)

[comment]: <> (│   │   └── user-profile)

[comment]: <> (│   │       ├── user-profile.component.css)

[comment]: <> (│   │       ├── user-profile.component.html)

[comment]: <> (│   │       ├── user-profile.component.spec.ts)

[comment]: <> (│   │       └── user-profile.component.ts)

[comment]: <> (│   ├── assets)

[comment]: <> (│   │   ├── css)

[comment]: <> (│   │   │   └── demo.css)

[comment]: <> (│   │   ├── img)

[comment]: <> (│   │   └── scss)

[comment]: <> (│   │       ├── core)

[comment]: <> (│   │       └── material-dashboard.scss)

[comment]: <> (│   ├── environments)

[comment]: <> (│   ├── favicon.ico)

[comment]: <> (│   ├── index.html)

[comment]: <> (│   ├── main.ts)

[comment]: <> (│   ├── polyfills.ts)

[comment]: <> (│   ├── styles.css)

[comment]: <> (│   ├── test.ts)

[comment]: <> (│   ├── tsconfig.app.json)

[comment]: <> (│   ├── tsconfig.spec.json)

[comment]: <> (│   └── typings.d.ts)

[comment]: <> (├── tsconfig.json)

[comment]: <> (├── tslint.json)

[comment]: <> (└── typings)

[comment]: <> (```)

[comment]: <> (## Browser Support)

[comment]: <> (At present, we officially aim to support the last two versions of the following browsers:)

[comment]: <> (<img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/chrome.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/firefox.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/edge.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/safari.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/opera.png" width="64" height="64">)



[comment]: <> (## Resources)

[comment]: <> (- Demo: <https://demos.creative-tim.com/material-dashboard-angular2/#/dashboard>)

[comment]: <> (- Download Page: <https://www.creative-tim.com/product/material-dashboard-angular2>)

[comment]: <> (- Documentation: <https://demos.creative-tim.com/material-dashboard-angular2/#/documentation/tutorial>)

[comment]: <> (- License Agreement: <https://www.creative-tim.com/license>)

[comment]: <> (- Support: <https://www.creative-tim.com/contact-us>)

[comment]: <> (- Issues: [Github Issues Page]&#40;https://github.com/creativetimofficial/material-dashboard-angular2/issues&#41;)

[comment]: <> (- [Material Kit]&#40;https://www.creative-tim.com/product/material-kit?ref=github-mda-free&#41; - For Front End Development)

[comment]: <> (## Reporting Issues)

[comment]: <> (We use GitHub Issues as the official bug tracker for the Material Dashboard. Here are some advices for our users that want to report an issue:)

[comment]: <> (1. Make sure that you are using the latest version of the Material Dashboard. Check the CHANGELOG from your dashboard on our [website]&#40;https://www.creative-tim.com/&#41;.)

[comment]: <> (2. Providing us reproducible steps for the issue will shorten the time it takes for it to be fixed.)

[comment]: <> (3. Some issues may be browser specific, so specifying in what browser you encountered the issue might help.)


[comment]: <> (## Technical Support or Questions)

[comment]: <> (If you have questions or need help integrating the product please [contact us]&#40;https://www.creative-tim.com/contact-us&#41; instead of opening an issue.)



[comment]: <> (## Licensing)

[comment]: <> (- Copyright 2018 Creative Tim &#40;https://www.creative-tim.com/&#41;)

[comment]: <> (- Licensed under MIT &#40;https://github.com/creativetimofficial/material-dashboard-angular2/blob/master/LICENSE.md&#41;)


[comment]: <> (## Useful Links)

[comment]: <> (- [More products]&#40;https://www.creative-tim.com/bootstrap-themes&#41; from Creative Tim)

[comment]: <> (- [Tutorials]&#40;https://www.youtube.com/channel/UCVyTG4sCw-rOvB9oHkzZD1w&#41;)

[comment]: <> (- [Freebies]&#40;https://www.creative-tim.com/bootstrap-themes/free&#41; from Creative Tim)

[comment]: <> (- [Affiliate Program]&#40;https://www.creative-tim.com/affiliates/new&#41; &#40;earn money&#41;)

[comment]: <> (##### Social Media)

[comment]: <> (Twitter: <https://twitter.com/CreativeTim>)

[comment]: <> (Facebook: <https://www.facebook.com/CreativeTim>)

[comment]: <> (Dribbble: <https://dribbble.com/creativetim>)

[comment]: <> (Google+: <https://plus.google.com/+CreativetimPage>)

[comment]: <> (Instagram: <https://www.instagram.com/CreativeTimOfficial>)

[comment]: <> ([CHANGELOG]: ./CHANGELOG.md)

[comment]: <> ([version-badge]: https://img.shields.io/badge/version-2.6.0-blue.svg)


CHANGELOG.md
[comment]: <> (## [2.6.0] - 2021-07-15)

[comment]: <> (### Updates)

[comment]: <> (- update to Angular 12)

[comment]: <> (- update all dependencies to match Angular 12 version)

[comment]: <> (When you build and serve your app it is possible that some warnings to appear on your terminal. Those will NOT affect your product.)

[comment]: <> (## [2.5.0] - 2020-10-07)

[comment]: <> (### Updates)

[comment]: <> (- update to Angular 10)

[comment]: <> (- update all dependencies to match Angular 10 version)


[comment]: <> (## [2.4.0] - 2020-03-05)

[comment]: <> (### Updates)

[comment]: <> (- update to Angular 9)

[comment]: <> (- update all dependencies to match Angular 9 version)

[comment]: <> (## [2.3.0] - 2019-06-11)

[comment]: <> (### Updates)

[comment]: <> (- update to Angular 8)

[comment]: <> (- update all dependencies to match Angular 8 version)

[comment]: <> (```)

[comment]: <> (@agm/core                           1.0.0-beta.5   →   1.0.0-beta.6)

[comment]: <> (@angular/animations                        7.0.2   →          8.0.0)

[comment]: <> (@angular/cdk                               7.0.2   →          8.0.1)

[comment]: <> (@angular/common                            7.0.2   →          8.0.0)

[comment]: <> (@angular/compiler                          7.0.2   →          8.0.0)

[comment]: <> (@angular/core                              7.0.2   →          8.0.0)

[comment]: <> (@angular/forms                             7.0.2   →          8.0.0)

[comment]: <> (@angular/http                              7.0.2   →         7.2.15)

[comment]: <> (@angular/material                          7.0.2   →          8.0.1)

[comment]: <> (@angular/platform-browser                  7.0.2   →          8.0.0)

[comment]: <> (@angular/platform-browser-dynamic          7.0.2   →          8.0.0)

[comment]: <> (@angular/platform-server                   7.0.2   →          8.0.0)

[comment]: <> (@angular/router                            7.0.2   →          8.0.0)

[comment]: <> (ajv                                        6.4.0   →         6.10.0)

[comment]: <> (bootstrap                                  4.1.0   →          4.3.1)

[comment]: <> (bootstrap-material-design                  4.1.1   →          4.1.2)

[comment]: <> (chartist                                  0.11.0   →         0.11.2)

[comment]: <> (core-js                                    2.4.1   →          3.1.3)

[comment]: <> (express                                   4.16.3   →         4.17.1)

[comment]: <> (googleapis                                28.1.0   →         40.0.0)

[comment]: <> (jquery                                     3.2.1   →          3.4.1)

[comment]: <> (moment                                    2.22.1   →         2.24.0)

[comment]: <> (perfect-scrollbar                          1.1.0   →          1.4.0)

[comment]: <> (popper.js                                 1.14.3   →         1.15.0)

[comment]: <> (rxjs                                       6.3.3   →          6.5.2)

[comment]: <> (rxjs-compat                                6.3.3   →          6.5.2)

[comment]: <> (zone.js                                   0.8.26   →          0.9.1)

[comment]: <> (@angular-devkit/build-angular              0.6.3   →        0.800.2)

[comment]: <> (@angular/cli                               7.0.2   →          8.0.2)

[comment]: <> (@angular/compiler-cli                      7.0.2   →          8.0.0)

[comment]: <> (@angular/language-service                  7.0.2   →          8.0.0)

[comment]: <> (@types/bootstrap                          3.3.32   →          4.3.0)

[comment]: <> (@types/chartist                           0.9.34   →         0.9.46)

[comment]: <> (@types/googlemaps                         3.30.8   →         3.36.4)

[comment]: <> (@types/jasmine                            2.5.38   →         3.3.13)

[comment]: <> (@types/jquery                            1.10.31   →         3.3.29)

[comment]: <> (@types/node                               6.0.73   →         12.0.7)

[comment]: <> (codelyzer                                  4.2.1   →          5.1.0)

[comment]: <> (jasmine-core                               3.3.0   →          3.4.0)

[comment]: <> (karma                                      2.0.0   →          4.1.0)

[comment]: <> (karma-cli                                  1.0.1   →          2.0.0)

[comment]: <> (karma-coverage-istanbul-reporter           1.4.2   →          2.0.5)

[comment]: <> (karma-jasmine                              1.1.1   →          2.0.1)

[comment]: <> (karma-jasmine-html-reporter                1.4.0   →          1.4.2)

[comment]: <> (protractor                                 5.3.1   →          5.4.2)

[comment]: <> (ts-node                                    5.0.1   →          8.2.0)

[comment]: <> (tslint                                     5.9.1   →         5.17.0)

[comment]: <> (typescript                                 3.1.6   →          3.4.5)

[comment]: <> (```)

[comment]: <> (### Bug fixing)

[comment]: <> (- browser console error)

[comment]: <> (```)

[comment]: <> (href="#pablo" -> href="javascript:void&#40;0&#41;")

[comment]: <> (href="#"      -> href="javascript:void&#40;0&#41;")

[comment]: <> (```)

[comment]: <> (- form-validation error from the `_forms.scss` file)

[comment]: <> (```)

[comment]: <> (@include form-validation-state&#40;"valid", $label-color&#41;; -> @include form-validation-state&#40;"valid", $label-color, $form-feedback-icon-valid&#41;;)

[comment]: <> (```)

[comment]: <> (- added hash for google url validation)

[comment]: <> (## [2.2.0] - 2018-11-14)

[comment]: <> (### Changes)

[comment]: <> (- update to Angular 7)

[comment]: <> (- update all dependencies to match Angular 7 version)

[comment]: <> (## [2.1.1] - 2018-05-23)

[comment]: <> (### Fixes)

[comment]: <> (- changed some links)

[comment]: <> (## [2.1.0] - 2018-04-27)

[comment]: <> (### Fixes)

[comment]: <> (- changed file structure)


src/app/dashboard/dashboard.component.ts
import { Component, OnInit, ViewChild } from '@angular/core';
import * as Chartist from 'chartist';
import { SharePointService } from '../shared/Services/share-point.service';
import { ToastrService } from 'ngx-toastr';
import { fromEvent } from 'rxjs';
import { map, debounceTime } from 'rxjs/operators';
import * as moment from 'moment';
import { AddUserDialogBoxComponent } from '../user/add-user/add-user-dialog-box.component';
import { MatDialog } from '@angular/material/dialog';
import { EditFileComponent } from './edit-file/edit-file.component';


@Component({
  selector: 'app-dashboard',
  templateUrl: './dashboard.component.html',
  styleUrls: ['./dashboard.component.css']
})
export class DashboardComponent implements OnInit {
  @ViewChild('myTable') table;
  @ViewChild('search', { static: false }) search: any;

  public role: any;
  public userId: any;
  pageSize: any;
  rowCount: any;
  curPage: any;
  public rows = [];
  public temp = [];

  constructor(public sharepointService: SharePointService, private toastr: ToastrService,
    public dialog: MatDialog) { }

  startAnimationForLineChart(chart) {
    let seq: any, delays: any, durations: any;
    seq = 0;
    delays = 80;
    durations = 500;

    chart.on('draw', function (data) {
      if (data.type === 'line' || data.type === 'area') {
        data.element.animate({
          d: {
            begin: 600,
            dur: 700,
            from: data.path.clone().scale(1, 0).translate(0, data.chartRect.height()).stringify(),
            to: data.path.clone().stringify(),
            easing: Chartist.Svg.Easing.easeOutQuint
          }
        });
      } else if (data.type === 'point') {
        seq++;
        data.element.animate({
          opacity: {
            begin: seq * delays,
            dur: durations,
            from: 0,
            to: 1,
            easing: 'ease'
          }
        });
      }
    });

    seq = 0;
  };
  startAnimationForBarChart(chart) {
    let seq2: any, delays2: any, durations2: any;

    seq2 = 0;
    delays2 = 80;
    durations2 = 500;
    chart.on('draw', function (data) {
      if (data.type === 'bar') {
        seq2++;
        data.element.animate({
          opacity: {
            begin: seq2 * delays2,
            dur: durations2,
            from: 0,
            to: 1,
            easing: 'ease'
          }
        });
      }
    });

    seq2 = 0;
  };
  ngOnInit() {
    this.role = localStorage.getItem('UserRole');
    this.userId = localStorage.getItem('UserId');
    this.fetch();
    /* ----------==========     Daily Sales Chart initialization For Documentation    ==========---------- */

    const dataDailySalesChart: any = {
      labels: ['M', 'T', 'W', 'T', 'F', 'S', 'S'],
      series: [
        [12, 17, 7, 17, 23, 18, 38]
      ]
    };

    const optionsDailySalesChart: any = {
      lineSmooth: Chartist.Interpolation.cardinal({
        tension: 0
      }),
      low: 0,
      high: 50, // creative tim: we recommend you to set the high sa the biggest value + something for a better look
      chartPadding: { top: 0, right: 0, bottom: 0, left: 0 },
    }

    var dailySalesChart = new Chartist.Line('#dailySalesChart', dataDailySalesChart, optionsDailySalesChart);

    this.startAnimationForLineChart(dailySalesChart);


    /* ----------==========     Completed Tasks Chart initialization    ==========---------- */

    const dataCompletedTasksChart: any = {
      labels: ['12p', '3p', '6p', '9p', '12p', '3a', '6a', '9a'],
      series: [
        [230, 750, 450, 300, 280, 240, 200, 190]
      ]
    };

    const optionsCompletedTasksChart: any = {
      lineSmooth: Chartist.Interpolation.cardinal({
        tension: 0
      }),
      low: 0,
      high: 1000, // creative tim: we recommend you to set the high sa the biggest value + something for a better look
      chartPadding: { top: 0, right: 0, bottom: 0, left: 0 }
    }

    var completedTasksChart = new Chartist.Line('#completedTasksChart', dataCompletedTasksChart, optionsCompletedTasksChart);

    // start animation for the Completed Tasks Chart - Line Chart
    this.startAnimationForLineChart(completedTasksChart);



    /* ----------==========     Emails Subscription Chart initialization    ==========---------- */

    var datawebsiteViewsChart = {
      labels: ['J', 'F', 'M', 'A', 'M', 'J', 'J', 'A', 'S', 'O', 'N', 'D'],
      series: [
        [542, 443, 320, 780, 553, 453, 326, 434, 568, 610, 756, 895]

      ]
    };
    var optionswebsiteViewsChart = {
      axisX: {
        showGrid: false
      },
      low: 0,
      high: 1000,
      chartPadding: { top: 0, right: 5, bottom: 0, left: 0 }
    };
    var responsiveOptions: any[] = [
      ['screen and (max-width: 640px)', {
        seriesBarDistance: 5,
        axisX: {
          labelInterpolationFnc: function (value) {
            return value[0];
          }
        }
      }]
    ];
    var websiteViewsChart = new Chartist.Bar('#websiteViewsChart', datawebsiteViewsChart, optionswebsiteViewsChart, responsiveOptions);

    //start animation for the Emails Subscription Chart
    this.startAnimationForBarChart(websiteViewsChart);
  }

  fetch() {
    this.sharepointService.getAllSharedFiles(this.role)
      .subscribe(success => {
        if (success) {
          this.rows = this.sharepointService.SharedFiles;
          this.temp = [...this.rows];
        }
      });
  }

  DownloadFile(row) {
    var link = row.Link;
    this.toastr.success("Download successfully", 'Success');

    window.open(link);
  }

  onEditDocument() {

  }
  ngAfterViewInit(): void {
    fromEvent(this.search.nativeElement, 'keydown')
      .pipe(
        debounceTime(550),
        map(x => {
          return x['target']['value']
        })
      )
      .subscribe(value => {
        this.updateFilter(value);
      });
  }

  updateFilter(val: any) {
    const value = val.toString().toLowerCase().trim();
    const count = 15;
    const keys = Object.keys(this.temp[0]);
    this.rows = this.temp.filter(item => {
      for (let i = 0; i < count; i++) {
        debugger;
        if (i != 5) {
          // check for a match
          if ((item[keys[i]] && item[keys[i]].toString().toLowerCase().indexOf(value) !== -1) || !value) {
            // found match, return true to add to result set
            return true;
          }
        } else {
          // check for a match
          if ((item[keys[i]] && moment(item[keys[i]]).format('DD-MM-YYYY').toString().toLowerCase().indexOf(value) !== -1) || !value) {
            // found match, return true to add to result set
            return true;
          }
        }
      }
    });

    this.table.offset = 0;
  }

  EditFile(row): void {
    const matDialogRef = this.dialog.open(EditFileComponent, {
      width: '900px',
      data: {
        ID: row.Id,
        Topic: row.Topic,
        Abstract: row.Abstract,
        Author: row.Author,
        Role: row.RoleName,
        Link: row.Link
      }
    });
  }

  DeleteFile(row): void {
    // Delete pop-up window code
    this.toastr.success("Shared File: " + row.Id + " delete successfully", 'Success');
  }
}
